_posts/2020-01-30-new-age-of-bastion.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: post
+title: "New age of Bastion"
+author: Mahmut Bulut
+---
+
+**Disclaimer:** This post is not only an announcement of new features of Bastion, which comes with 0.3.4; it also declares our future perspective for the project.
+
+So we had a long way since our project had started in June 2019. Right now, we are evolving from a runtime that enables fault-recovery for your server-side applications to an agnostic runtime, which has its executor but also works with other executors. Since the beginning we can interoperate with [async-std](https://async.rs) without the hassle.
+
+We have started with an idea of runtime, which can recover from failures in the application domain, no matter what happens. Async/await landed, and we got stronger and well known in the community. Then Bastion deployed to AWS lambdas, did, and still doing its job in data processing pipelines. Benchmarks of sparse queries outperformed plenty of sparse queries and aggregations over NYC taxi data in various inmem benchmarks.
+
+Where futures-executor performs:
+```
+test count_aggregation      ... bench: 108,916,527 ns/iter (+/- 11,909,528)
+```
+Bastion performed:
+```
+test count_aggregation      ... bench: 87,096,485 ns/iter (+/- 3,874,812)
+```
+All have started with an idea of replicating what Erlang does its runtime. Using the runtime for the data processing comes right after it. Writing a generic runtime for fault-tolerant middleware, distributed systems, and data processing pipelines was the main goal of Bastion.
+
+I didn't aim for faster runtime, but I aimed to exploit the whole resources whenever it's possible. Bastion also doesn't aim for a being replacement to runtimes out there. It just unfolds a different approach to design systems. This system design is very graspable for the people who come from a functional background like Scala, Erlang, or Haskell. When we released the Bastion 0.1 in June 2019 with Tokio runtime, I realized that it's not the way that this project can continue. With premature design assumptions of the runtime, I struggled with developing the project with Tokio runtime. So we had a multiple run queue based statistics sampled async runtime. Throughout time, people wanted to use blocking tasks also. So we came with an idea of using an adaptive blocking pool, which can [automatically scale based on the throughput](https://docs.rs/bastion-executor/0.3.4/bastion_executor/blocking/index.html) and its' hogs up and down easily.
```

Comment:
aim for a being replacement to runtimes out there.
aim to replace every other existing runtimes.

When we released the Bastion 0.1 
When we released Bastion 0.1

With premature design assumptions of the runtime, I struggled with developing the project with Tokio runtime. 
Could we elaborate a bit or be a bit more specific ? I feel like it's a bit mean towards tokio, whereas the actual reason is that you eventually figured out your goal isnt what tokio provides ? What exactly is the goal tokio couldnt achieve? 

So we had a multiple run queue based statistics sampled async runtime.
=> i m sorry i dont understand :/

Throughout time, people wanted to use blocking tasks also.
Over time, people wanted to be able to run blocking tasks as well as non blocking ones.

and its' hogs up and down easily.
=> do we mean we can upscale / downscale easily ?